src/hyperion/experiment_plans/tests/run_engine_simulator.py

```diff
@@ -0,0 +1,161 @@
+        """
+        self.message_handlers.append(
+            MessageHandler(
+                lambda msg: msg.command == "wait"
+                and (group == "any" or msg.kwargs["group"] == group),
+                handler,
+            )
+        )
+
+    def fire_callback(self, document_name, document):
+        """Fire all the callbacks registered for this document type in order to simulate something happening
+        Args:
+             document_name: document name as defined in the Bluesky Message Protocol 'subscribe' call,
+             all subscribers filtering on this document name will be called
+             document: the document to send
+        """
+        for callback_func, callback_docname in self.callbacks.values():
+            if callback_docname == "all" or callback_docname == document_name:
+                callback_func(document_name, document)
+
+    def simulate_plan(self, gen: Generator[Msg, object, object]) -> list[Msg]:
+        """Simulate the RunEngine executing the plan
+        Args:
+            gen: the generator function that executes the plan
+        Returns:
+            a list of the messages generated by the plan
+        """
+        messages = []
+        send_value = None
+        try:
+            while msg := gen.send(send_value):
+                send_value = None
+                messages.append(msg)
+                LOGGER.debug(f"<{msg}")
+                if handler := next(
+                    (h for h in self.message_handlers if h.predicate(msg)), None
+                ):
+                    send_value = handler.runnable(msg)
+
+                if send_value:
+                    LOGGER.debug(f">send {send_value}")
+        except StopIteration:
+            pass
+        return messages
+
+    def _add_callback(self, msg_args):
+        self.callbacks[self.next_callback_token] = msg_args
+        self.next_callback_token += 1
+
+
+def add_simple_pin_tip_centre_handlers(sim: RunEngineSimulator):
+    """Handlers to simulate a basic fake pin tip"""
+    sim.add_handler(
+        ("trigger", "read"),
+        "oav_mxsc_pin_tip",
+        lambda msg: {"oav_mxsc_pin_tip_triggered_tip": {"value": (100, 100)}},
+    )
+    sim.add_handler(
+        "read",
+        "oav_mxsc_top",
+        lambda msg: {"values": {"value": [50, 51, 52, 53, 54, 55]}},
+    )
+    sim.add_handler(
+        "read",
+        "oav_mxsc_bottom",
+        lambda msg: {"values": {"value": [50, 49, 48, 47, 46, 45]}},
+    )
+
+
+def add_simple_oav_mxsc_callback_handlers(sim: RunEngineSimulator):
+    """Handlers to simulate a basic oav callback firing"""
+    sim.add_handler(
+        "set",
+        "oav_mxsc_enable_callbacks",
+        # XXX what are reasonable values for these?
+        lambda msg: sim.fire_callback(
+            "event",
+            {
+                "data": {
+                    "oav_snapshot_last_saved_path": "xxx",
```

Comment:
these can probably just be something like `"/tmp/image1.png"`, `"/tmp/image2.png"`, `"/tmp/image3.png"`